lib/request.js

```diff
@@ -64,10 +64,17 @@ function Request (options) {
     this.headers.cookie = cookieValues.join('; ')
   }
 
-  this.connection = {
+  this.socket = {
     remoteAddress: options.remoteAddress || '127.0.0.1'
   }
-  this.socket = this.connection
+
+  Object.defineProperty(this, 'connection', {
+    get () {
+      // if node major ≥13 emit a warning with https://github.com/fastify/fastify-warning
```

Comment:
The comment here is not sufficient. An actual test against against `process.version` is needed along with a warning issued through the interface that `fastify-warning` provides.

Reply:
Understood.  Maybe something like https://github.com/fastify/fastify-plugin/blob/d3b2ca47404e622f7c474493bb0624bf95a2e4f4/test/esm/index.test.js#L6, and then use fastify-warning to actually log that message?

Reply:
Correct. Fastify itself uses `fastify-warning` to issue deprecation notices. You can learn from there as well.

Reply:
Great, thanks @jsumners, this is really helpful.  I'll dig around and update this to do it correctly.